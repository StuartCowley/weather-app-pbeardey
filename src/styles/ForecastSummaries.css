/* src/styles/ForcastSummaries.css */

.forecast-summaries {
    display: flex;
    flex-wrap: wrap;
    justify-content: center;

}