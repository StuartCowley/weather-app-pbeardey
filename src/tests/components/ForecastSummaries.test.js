// src/tests/components/ForecastSummaries.test.js

import React from "react";
import { render, screen } from "@testing-library/react";
import ForecastSummaries from "../../components/ForecastSummaries";

describe("ForecastSummaries", () => {
  const validProps = [
    {
      date: 1111111,
      description: "Stub description",
      icon: "800",
      temperature: {
        min: 12,
        max: 22,
      },
    },
    {
      date: 26311111,
      description: "Stub2 description",
      icon: "802",
      temperature: {
        min: 15,
        max: 25,
      },
    },
  ];
  function onForecastSelect() {}

  describe("ForecastSummaries-snapshot", () => {
    it("renders correctly", () => {
      const { asFragment } = render(
        <ForecastSummaries
          forecasts={validProps}
          onForecastSelect={onForecastSelect}
        />
      );

      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe("ForecastSummaries-values", () => {
    it("renders 2 forecasts", () => {
      const { getAllByTestId } = render(
        <ForecastSummaries
          forecasts={validProps}
          onForecastSelect={onForecastSelect}
        />
      );
      expect(getAllByTestId("forecast-summary")).toHaveLength(2);
    });

    it("renders 2 buttons", () => {
      render(
        <ForecastSummaries
          forecasts={validProps}
          onForecastSelect={onForecastSelect}
        />
      );
      const buttons = screen.getAllByRole("button");

      expect(buttons).toHaveLength(2);
    });
  });
});
