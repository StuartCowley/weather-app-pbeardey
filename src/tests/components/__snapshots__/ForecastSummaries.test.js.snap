// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummaries ForecastSummaries-snapshot renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-summaries"
  >
    <div
      class="forecast-summary"
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary__date"
      >
        Thu Jan 01 1970
      </div>
      <div
        class="forecast-summary__icon"
        data-testid="forecast-icon"
      >
        <i
          class="wi wi-day-sunny wi-flip-horizontal wi-rotate-90"
          name="owm"
        />
      </div>
      <div
        class="forecast-summary__temperature"
      >
        22°C
      </div>
      <div
        class="forecast-summary__description"
      >
        Stub description
      </div>
      <button
        type="button"
      >
        More details
      </button>
    </div>
    <div
      class="forecast-summary"
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary__date"
      >
        Thu Jan 01 1970
      </div>
      <div
        class="forecast-summary__icon"
        data-testid="forecast-icon"
      >
        <i
          class="wi wi-day-cloudy wi-flip-horizontal wi-rotate-90"
          name="owm"
        />
      </div>
      <div
        class="forecast-summary__temperature"
      >
        25°C
      </div>
      <div
        class="forecast-summary__description"
      >
        Stub2 description
      </div>
      <button
        type="button"
      >
        More details
      </button>
    </div>
  </div>
</DocumentFragment>
`;
